src/ggml.c

```diff
@@ -12206,37 +12206,34 @@ static void ggml_compute_forward_upscale(
 static void ggml_compute_forward_pad_f32(
     const struct ggml_compute_params * params,
     const struct ggml_tensor * src0,
-    struct ggml_tensor * dst) {
+          struct ggml_tensor * dst) {
 
     if (params->type == GGML_TASK_INIT || params->type == GGML_TASK_FINALIZE) {
         return;
     }
 
     GGML_ASSERT(src0->nb[0] == sizeof(float));
-    GGML_ASSERT(dst->nb[0] == sizeof(float));
+    GGML_ASSERT( dst->nb[0] == sizeof(float));
 
     const int ith = params->ith;
     const int nth = params->nth;
 
     GGML_TENSOR_UNARY_OP_LOCALS
 
-    const int padding_factor = dst->op_params[0];
+    float * dst_ptr = (float *) dst->data;
+
+    // TODO: optimize
 
-    float* src0_ptr = (float*)src0->data;
-    float* dst_ptr = (float*)dst->data;
+    for (int64_t i2 = 0; i2 < ne2; ++i2) {
+        for (int64_t i1 = ith; i1 < ne1; i1 += nth) {
+            for (int64_t i0 = 0; i0 < ne0; i0++) {
+                for (int64_t i3 = 0; i3 < ne3; i3++) {
+                    const int64_t dst_idx = i3*(ne0*ne1*ne2) + i2*(ne0*ne1) + i1*ne0 + i0;
 
-    const so2 = ne00 * ne01;
-    const so3 = ne00 * ne01 * ne02;
-    const do2 = ne0 * ne1;
-    const do3 = ne0 * ne1 * ne2;
+                    const float * src_ptr = (float *)((char *) src0->data + i3*nb03 + i2*nb02 + i1*nb01 + i0*nb00);
 
-    for (int j = 0; j < ne2; j ++) {
-        for (int k = ith; k < ne1; k += nth) {
-            for (int l = 0; l < ne0; l++) {
-                for (int i = 0; i < ne3; i++) {
-                    int dst_idx = i * do3 + j * do2 + k * ne0 + l;
-                    if (l < ne00 && k < ne01  && j < ne02 && i < ne03) {
-                        dst_ptr[dst_idx] = src0_ptr[i * so3 + j * so2 + k * ne00 + l];
+                    if (i0 < ne00 && i1 < ne01 && i2 < ne02 && i3 < ne03) {
```

Comment:
Try to follow the convention for the indices

src0 -> i00, i01, i02, i03
src1 -> i10, i11, i12, i13
dst. -> i0,  i1,  i2,  i3

Same for number of elements (`ne`) and strides (`nb`)

P.S. please double-check that I didn't break this function - I don't have a test atm